e107_plugins/forum/shortcodes/batch/forum_shortcodes.php

```diff
@@ -479,6 +479,8 @@ function sc_lastpost($parm = null)
 		$urlData        = array('forum_sef'=>$this->var['forum_sef'], 'thread_id'=>$lastpost['post_thread'],'thread_sef'=>$lastpost['thread_sef']);
 		$url            = e107::url('forum', 'topic', $urlData)."?last=1#post-".$lastpost['post_id'];
 		$lastpost_username = empty($this->var['user_name']) ? e107::getParser()->toHTML($this->var['forum_lastpost_user_anon']) : "<a href='".e107::url('user/profile/view', array('name' => $this->var['user_name'], 'id' => $this->var['forum_lastpost_user']))."'>{$this->var['user_name']}</a>";
+		$shortDate = e107::getParser()->toDate($lastpost_datestamp,'short');
+		$longDate = e107::getParser()->toDate($lastpost_datestamp,'long');
 		$relativeDate = e107::getParser()->toDate($lastpost_datestamp,'relative');
 

```

Comment:
This will work, but you're running toDate() 3 times to produce one date output. I'll add something shortly.